### ISSUE TITLE
Can you please add llava-phi-3-mini by xtuner?

### ISSUE THREAD

**Issue by yashasnadigsyn** [ollama/ollama#3929]
Here is the model gguf link: https://huggingface.co/xtuner/llava-phi-3-mini-gguf

Here is the model hf link: https://huggingface.co/xtuner/llava-phi-3-mini-hf

I have been trying to add it manually by modelfile but i can't seem to understand the template. I tried using llava template, bakllava template, other multimodal templates but the model confuses. 

Can anyone help me?

**Comment by TimeLordRaps** [comment]
I've tried adding it using the split FROM logic added in #1308 that works
I've tried concatenating both files as was suggested in that thread and that works.
Here's the most recent modelfile I am using:
```
FROM "\path\to\ggml-model-int4.gguf"
FROM "\path\to\mmproj-model-f16.gguf"
TEMPLATE """{{ if .System }}<|system|>
{{ .System }}<|end|>
{{ end }}{{ if .Prompt }}<|user|>
{{ .Prompt }}<|end|>
{{ end }}<|assistant|>
{{ .Response }}<|end|>
"""
PARAMETER stop "<|user|>"
PARAMETER stop "<|assistant|>"
PARAMETER stop "<|system|>"
PARAMETER stop "<|end|>"
PARAMETER stop "<|endoftext|>"

PARAMETER num_keep 4
PARAMETER num_ctx 4096

```

I will note ocr is fairly weak in comparison to what I'd expect from a better projector so might be worth the wait, at least in prelim tests on default ollama temp.

**Comment by yashasnadigsyn** [comment]
Thank you so much. I was an idiot for even forgetting the mmproj file. I wasted the whole day yesterday just tinkering with the int4 file and forgot the mmproj file. The modelfile works perfectly. Thank you for it.

Also for the OCR, I have been able to get somewhat good results with "demanding" prompts.

This was my prompt for the same above image:
>  What are the texts in the image. Your output should only have the texts in the image without any other things.

The output:
> Ollama Vision Welcome to the era of open-source multiplatform models. Open-source models can be made easily now.

But, using the same prompt again:
>  What are the texts in the image. Your output should only have the texts in the image without any other things.

I get bad results: 
> Ollama Vision Welcome to the era of open-source multiplatform models. Operate web browsers and make code available now.

But, it is good with object recognition. Personally, i found it faster and better than llava-7b-v1.6-mistral and bakllava-7b-v1 though i have tested it with only two images. 

For OCR especially, I have been looking at this model by internlm: [internlm-xcomposer2-vl-1.8b](https://huggingface.co/internlm/internlm-xcomposer2-vl-1_8b). It has pretty good results.

Again, Thank you so much for the modelfile.
